bindings/java/mongocrypt/build.gradle.kts

```diff
@@ -135,7 +135,7 @@ val jnaMappingList: List<LibMongoCryptS3Data> = listOf(
         LibMongoCryptS3Data("rhel-67-s390x", "linux-s390x"),
         LibMongoCryptS3Data("ubuntu1604-arm64", "linux-aarch64"),
         LibMongoCryptS3Data("windows-test", "win32-x86-64"),
-        LibMongoCryptS3Data("macos", "darwin")
```

Comment:
To clarify the Java library will only package the `macos_x86_64` bindings for Mac users.

Reply:
Yes.

The `publish-snapshot` task depends on the `build-and-test-java` task running on the `macos-1014` distro. The `macos-1014` distro only publishes `x86_64` builds.

The description of https://github.com/mongodb/libmongocrypt/pull/234 includes an attempted patch build with running `build-and-test-java` on the `macos-1100-arm64`. The `build-and-test-java` task failed with what looked like [Java environment issue](https://spruce.mongodb.com/task/libmongocrypt_macos_m1_build_and_test_java_patch_f4fcabfca2873a73c841556eed0e4a7370a2dfba_61df38b73e8e8655c0380c22_22_01_12_20_23_19/logs?execution=0).

I decided to keep the current behavior. Have `publish-snapshot` depend on the `build-and-test-and-java` run on the `macos-1014`

Reply:
:+1: